src/app/clusters/thermostat-server/thermostat-server-suggestions.cpp

```diff
@@ -38,7 +39,76 @@ namespace app {
 namespace Clusters {
 namespace Thermostat {
 
-extern ThermostatAttrAccess gThermostatAttrAccess;
+CHIP_ERROR RemoveExpiredSuggestions(Delegate * delegate)
+{
+    VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    uint32_t currentMatterEpochTimestampInSeconds = 0;
+    CHIP_ERROR err                                = System::Clock::GetClock_MatterEpochS(currentMatterEpochTimestampInSeconds);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+
+    for (int i = static_cast<int>(delegate->GetNumberOfThermostatSuggestions() - 1);
+         i >= 0 && delegate->GetNumberOfThermostatSuggestions() > 0; i--)
```

Comment:
So this is making some assumptions about how removal is going to be order-preserving, right?  Does anything in the delegate API guarantee that?

Past that, I don't understand what the `delegate->GetNumberOfThermostatSuggestions() > 0` check is there for.  Say we start with `delegate->GetNumberOfThermostatSuggestions() == 4`.  We check the thing at index 3, it's expired we remove it.  Now `i` becomes 2, but say `delegate->GetNumberOfThermostatSuggestions()` becomes ` for some reason (so > 0 tests true).  What is the > 0 check buying us in this situation?

Reply:
done.